### DISCUSSION TITLE
need inline train protection devices. Also for Eurobalise and balise group

### DISCUSSION THREAD

**Discussion by sixdiamants** [railtoolkit/tikz-trackschematic#28] [General]
The eurobalise and balisegroup are in high demand. 

I considered using the `transmitter` symbol but that doesn't fit the bill because it seems to be designed with a Gleismagnet in mind and is placed next to the track. 

I suggest introducing a simple rectangle shape to represent a **generic train protection spot device**. This could cover other devices like french crocodile. The (euro-)balise is considered such a device.

A balise group could be represented by a grouping BG4711 of balises or BG4712 when the individual balises aren't represented 
![image](https://user-images.githubusercontent.com/34092440/152679334-9e5d0f4f-769c-4e89-a1ef-cb7d02c38da0.png)
Note that Individual eurobalises have no orientation but the balisegroup does.

A balise group could be given the generic name "information point"; as it is known in France and Italy and other countries that have used balises for a long time. Thus, a eurobalise and balisegroup is just a special kind of information point. 

[Here are some supporting diagrams](https://dataprep.eulynx.eu/2022-01/EARoot/EA2/EA4/EA4/EA7250.htm). 

Finally, there's a need to represent a switchable balise. ProRail uses below symbols where the left one has got a little cable attached to indicate that it is a switched balise. You could do something similar.

![image](https://user-images.githubusercontent.com/34092440/152679454-29c76f56-ea2e-4577-a8d2-ec56669b0719.png)

**Comment by kaat0** [comment]
The transmitter symbols definitely need improvement!

Would it help to put the rectangle of the transmitter in the middle of the track?
![transmitter](https://user-images.githubusercontent.com/142348/152702260-1f34b9a6-e59f-4197-850e-1161f855415c.png)

With the current balise symbol, however, I see a risk of confusion with routes and driving relationships. In general, a triangle with a line is more of an arrow and thus a direction than an object.

In your example, I also see the danger of confusing BG4712 with a ghost train.

I suggest that we first collect the information that needs to be represented by a balise symbol.

**Comment by sixdiamants** [comment]
I think rather little graphical information is needed associated with a balise. We may want to steer clear of trying to visualise an application direction because it confuses people: balise group direction is given by B0-B1-B2 but a message *inside* the balise can act in normal or reverse direction so the meaning of application direction is equivocal.  But one still wants to represent that a balise group consists of more than one balise and which is the first one.


Your first picture, a balise group as a box with internal compartments is pretty good. 
This said, for many applications, suffice to represent a balise group as an "information point".
I think that DB draws simple boxes to represent balises.

Summarising the requirements:
- visualise the balise group and the extremity where the zero-balise is placed.
- visualise the number of balises in the group
- show which, if any, balise is switchable

I propose this:

![image](https://user-images.githubusercontent.com/34092440/152845555-eee44da5-78c4-468e-bf65-5126f906bd19.png)

**Comment by kaat0** [comment]
Is the larger shape of the single and double balise intentional? I don't find it problematic to keep the size of an individual balise constant.
For a size comparison with the other symbols and for later implementation, here is a suggestion:
![transmitter2_size](https://user-images.githubusercontent.com/142348/153035678-394e0cc6-5452-4cce-8239-110d9591a4f2.png)

With the switchable balises, I find the line problematic because of the size and recognisability.
However, I don't have a better suggestion. I tried a number of monochromatic options for a single balise:
![transmitter2](https://user-images.githubusercontent.com/142348/153035701-046a804a-0643-40a6-8933-e3820083fb64.png)

what do you find particularly bad?

**Comment by sixdiamants** [comment]
DB and others draw the single balise as a square though in reality it is a rectangle. 

During design, people may pencil in a balise group without knowledge of the number of balises inside. That's why I opt for the rectangle symbol with the "heavy side" representing the origin.

wikipedia suggests fixed vs. transparent Eurobalise:
>a balise may be either a '**Fixed** Data Balise,' or 'Fixed Balise' for short, transmitting the same data to every train, or a '**Transparent** Data Balise' which transmits variable data, also called a 'Switchable' or 'Controllable Balise'

from which I deduce that a window-like shape suits the transparent balise whereas a trellis would suit the fixed balise.
from left to right
transparent balise, fixed balise, balise group with origin on the left hand side, balise group consisting of a transparent and fixed balise.
Howzat ?

**Comment by kaat0** [comment]
After a lengthy discussion with my colleagues, it is now important to me that the direction in which the balise acts is recognizable from the depiction.

Here is a new proposal with fixed and switchable balises as well as unquantified and with an exact number of balises:
![transmitter3](https://user-images.githubusercontent.com/142348/153768968-d95adcf6-9e63-4b2b-92b3-04ef9bf5e8f7.png)

**Comment by sixdiamants** [comment]
I don't think that the individual balise has an application direction; it's the track-to-train messages inside the balise that have an application direction. There can be a mix of messages in a balise/BG and messages can be toggled on/off.

The balise group *does* have an orientation, the wording "orientation" instead of application direction expresses the notion that this orientation, captured by NID_PIG, lets the train establish its direction of movement. I.e. normal *with respect to the BG-orientation* when it passes NID_PIG=0,1,2... and reverse otherwise. So a train can decide that "the message in this BG has application direction reverse and I ignore it because I'm traversing the BG in normal direction".

The drawing on the top right-hand side of [the diagram on this page](https://dataprep.eulynx.eu/2022-01/EARoot/EA2/EA4/EA4/EA7250.htm) sums it all up. This was inspired by the SNCF way of representing balises and packets in the balises. Note that the drawing represents the balise group as an oriented triangle for the reason mentioned before.

I question why a balise should be placed to the right/left of track; it's physically at the centre and has no application direction.

In conclusion, I would welcome symbols that are 

- centered on the track
- are transparant/opaque to show that they're switched (transparent!) or fixed
- visualise orientation of the balise group 
 - I suggest using the triangle for a BG with unquantified number of balises
 - marking the Position-in-group when the balises in a BG are displayed (P0, P1 below but I'm not sure this is a good idea).
![image](https://user-images.githubusercontent.com/34092440/154260953-7a6904a9-4270-4026-b21c-b3e4d076686d.png)

**Comment by kaat0** [comment]
1. **the triangle for unquantified numbers**: As stated earlier: With a triangle, I see a risk of confusion with routes and driving relationships. In general, a triangle with a line is more of an arrow and thus a direction than an object. Also, the overall composition for easier perception becomes more complicated with other objects currently using the triangle (route, distant signal, speed signal). Furthermore, it is not wise to use two geometrically different symbols for essentially the same object.
To sum it up: let us stick with the rectangle/square for all balises.

2. **application direction**: The application direction needs to be addressed. I do not think, that doing it by a label is sufficient, due to the break in perceiving text versus perceiving symbols. Different options are needed since the triangle is problematic as well (see section 1.). The track schematic library has a built-in meaning of direction, which results from the left or right-hand driving of the railways. The left or right-hand driving can be seen in the signals and is also used for other symbols (clearing points, train berth). For example, in the case of right-hand driving, elements that are placed below the track are valid for the train travelling forward. And elements that are valid for a train travelling in the opposite direction are placed on top of the track. This also applies to the current transmitters and they are not designed with a _Gleismagnet_ (track magnet) in mind which is placed next to the track. It just happens that the technology to get application direction to a _Gleismagnet_ results also in placement left or right of a track. A track schematic must be abstract in order to represent the crucial information. Personally, I don't find it crucial that a balise, unlike a _Gleismagnet_, is placed in the middle of the track. abstractly, however, it is important that there is a sense of direction that can be easily recognised. On the abstract level of information transmission from infrastructure to vehicle, balises and _Gleismagnet_ do not differ. Technically they are worlds apart, but not in terms of meaning. In the case of signals, the technical zoo of signal aspects is not shown because it would simply be too much. The fact that a single balise has no direction can be compensated by displaying it on both sides of the track.
But I do recognise the problem to reach a universal symbol with many different already out there. And I am unsure how to proceed on the application direction.

3. **transparent/opaque vs. fixed**: I understand that the distinction between transparent and opaque comes from electrical engineering, but I would be very reluctant to apply this figurative language directly to the symbols. It could also be argued that with a black track, a "transparent" balise would also have to be black. with the image that a fixed balise is simple and something is added with a switchable balise, you arrive at the opposite representation. In addition, a large part of the transport science community that uses such track schematics on a larger scale does not consist mainly of electrical engineers. Therefore, the use of the wording "fixed" for a black rectangle with a white background and "switched" with an additional black square seems better.

I want to thank you for providing very helpful input. Do you have any ideas on how we can get a result that makes sense? Maybe I need to add the option for local flavour sooner.

**Comment by kaat0** [comment]
You probably ment this drawing:
![transmitter_eulynx](https://user-images.githubusercontent.com/142348/154742731-72113be2-115b-4ef9-9ac4-a3127d11c09a.png)

Currently, I would picture it in the trackschematic this way:
![transmitter_eulynx2](https://user-images.githubusercontent.com/142348/154743221-332ad8af-62da-4c03-bff2-b01d60d8131e.png)

**Comment by sixdiamants** [comment]
We're reaching consensus. I think it is important to express two directions to cater for the different use cases
- BG orientation expressed by M_PIG=0..n (as shown by the yellow triangle in the EULYNX drawing above)
- application direction of track to train packets (as expressed by the arrow over P27, P3, P12...)

I suggest that the balise group's label express the BG orientation.  I prefer Label BG03 because it expresses the BG orientation more strongly than BG01. Just for the record, ETCS defines orientation of single-balise "groups" as well. 

![image](https://user-images.githubusercontent.com/34092440/155946457-eed2ec26-c2cd-4954-ab98-12c3dd4a7371.png)

**Comment by kaat0** [comment]
I would always arrange the labels in the same wayiIn order to treat the BG orientation and the direction of application equally - even if it is only a single balise.
It would look like this:
![transmitter4](https://user-images.githubusercontent.com/142348/156020948-f5041f41-2fd0-4c25-ae2c-e58e2d16fa19.png)

**Comment by kaat0** [comment]
included with d1ff03de711b471efa18c06f7c7aac061d95cf53
